src/Main.py

```diff
@@ -113,18 +80,16 @@
                         mime="text/json",
                     )
             except openai.RateLimitError as e:
-                st.markdown(
-                    "It looks like you do not have OpenAI API credits left. Check [OpenAI's usage webpage for more information](https://platform.openai.com/account/usage)"
-                )
+                st.markdown("It looks like you do not have OpenAI API credits left. Check [OpenAI's usage webpage for more information](https://platform.openai.com/account/usage)")
                 st.write(e)
             except openai.NotFoundError as e:
-                st.warning(
-                    "It looks like you do not have entered you Credit Card information on OpenAI's site. Buy pre-paid credits to use the API and try again.",
-                    icon="💳"
-                )
+                st.warning("It looks like you do not have entered your Credit Card information on OpenAI's site. Buy pre-paid credits to use the API and try again.", icon="💳")
                 st.write(e)
             except Exception as e:
                 st.error("An error occurred while generating the resume. Please try again.")
                 st.write(e)
     else:
         st.info("Please upload a file to get started.")
+
+if __name__ == '__main__':
+    main()
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rohankokkulabito/resfolio/pull/2#issuecomment-2225210660>#c0d9f6</a><span> - 07/12/2024, 09:44 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'main()' function is being called without any error handling at the top level. If an unhandled exception occurs within 'main()', it could cause the entire application to crash.
</div>
<div id="fix">
<b>Fix:</b> Wrap the 'main()' function call in a try-except block to catch any unhandled exceptions and log them appropriately. This will ensure that the application does not crash unexpectedly and provides useful error information.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -93,3 +93,7 @@
  if __name__ == '__main__':
 -    main()
 +    try:
 +        main()
 +    except Exception as e:
 +        st.error("An unexpected error occurred. Please try again later.")
 +        st.write(e)
```
</div>
</details>
</div>
<br>


</div>